docs: Implement the first version of the new docs (#160)
invertase/globe: docs/infrastructure/overview/resources-and-limits.mdx

---
title: Resources & Limits
---

Globe runs your applications on a globally distributed infrastructure designed for automatic **routing**, **caching**, and **scaling**. To maintain high reliability and platform stability, we apply resource usage limits based on fair use principles.

Understanding these limits helps you design resilient applications and avoid unexpected disruptions.

## Default Resource Allowances

Your Globe account operates with the following default resource thresholds and behaviours:

| **Resource**       | **Current Thresholds**    | **Notes**                                                    |
| ------------------ | ------------------------- | ------------------------------------------------------------ |
| **Requests**       | 1 million per month       | All HTTP requests count toward this limit                    |
| **Bandwidth**      | 10GB per month (uncached) | Static assets served from cache don't count toward bandwidth |

> | **Memory**         | 256MB per container       | Applies per running instance of your application             |

Reviewer: 512 MB per container

## Requests

Your Globe account includes **1 million requests per month**. Every HTTP interaction with your app counts toward this total, including:

- API endpoint calls
- Page views rendered by your application
- Requests for static assets (like images or scripts), _even if served from Globe's edge cache_
- Each new WebSocket connection established

When your application reaches this request limit, any new incoming requests will be **rejected** with an error response until your usage cycle resets at the start of the next month.

## Bandwidth

Your account includes **10GB of uncached bandwidth** per month. Bandwidth usage is calculated based on the total size of:

- Response payload
- Request payload
- HTTP headers

Static assets served directly from Globe's optimized edge cache **do not** count against your bandwidth limit — but the initial request to fetch or revalidate a cached asset **does** count toward your requests.

If you exceed your bandwidth allowance, you may experience request rejections or slower service. Bandwidth usage resets monthly.

## Memory Usage

> Each running instance of your application in Globe has access to **256MB of memory**.

Reviewer: 512 MB

- API endpoint calls
- Page views rendered by your application
- Requests for static assets (like images or scripts), _even if served from Globe's edge cache_
- Each new WebSocket connection established

When your application reaches this request limit, any new incoming requests will be **rejected** with an error response until your usage cycle resets at the start of the next month.

## Bandwidth

Your account includes **10GB of uncached bandwidth** per month. Bandwidth usage is calculated based on the total size of:

- Response payload
- Request payload
- HTTP headers

Static assets served directly from Globe's optimized edge cache **do not** count against your bandwidth limit — but the initial request to fetch or revalidate a cached asset **does** count toward your requests.

If you exceed your bandwidth allowance, you may experience request rejections or slower service. Bandwidth usage resets monthly.

## Memory Usage

Each running container in your Globe deployment has access to **256MB of memory**. This memory is shared across:

- Your application code
- Libraries, packages, and dependencies
- In-memory caches
- Active connections and stateful data

> If your application exceeds this memory limit, Globe may **terminate the container** to maintain platform stability. Any active requests handled by that container will fail as a result.

Reviewer 2: ```suggestion
If your application exceeds this memory limit, Globe will terminate your application instance. Also, any active requests will fail as a result.
```

## CPU Usage

> Globe dynamically monitors CPU load across the platform. There's no hard limit per container instead, applications using excessive CPU for extended periods may have requests **terminated** to protect overall system performance.

Reviewer 2: ```suggestion
Globe dynamically monitors CPU load across the platform. There's no hard limit per instance instead, applications using excessive CPU for extended periods may have requests **terminated**.
```

## Execution Time

> Each request to your application is allowed a maximum of **10 seconds** from start to finish, from the time Globe receives the request to when the response is completed.

Reviewer 2: ```suggestion
Each request to your application is allowed a maximum of **30 seconds** from start to finish, from the time Globe receives the request to when the response is completed.
```

> If a request takes longer than 10 seconds, it will be **automatically terminated**, regardless of its progress.

Reviewer 2: ```suggestion
If a request takes longer than 30 seconds, it will be **automatically terminated**, regardless of its progress.
```
